diff --git a/django/db/migrations/operations/models.py b/django/db/migrations/operations/models.py
index c120f5d32d..1742360a0c 100644
--- a/django/db/migrations/operations/models.py
+++ b/django/db/migrations/operations/models.py
@@ -528,6 +528,13 @@ class AlterTogetherOptionOperation(ModelOptionOperation):
     def migration_name_fragment(self):
         return 'alter_%s_%s' % (self.name_lower, self.option_name)
 
+    def reduce(self, operation, app_label):
+        return (
+            super().reduce(operation, app_label) or
+            self.name_lower == getattr(operation, 'name_lower', None) and
+            not self.option_value
+        )
+
 
 class AlterUniqueTogether(AlterTogetherOptionOperation):
     """
