diff --git a/sympy/parsing/latex/lark/grammar/latex.lark b/sympy/parsing/latex/lark/grammar/latex.lark
index be4d0c74601b..f8879a31a9b2 100644
--- a/sympy/parsing/latex/lark/grammar/latex.lark
+++ b/sympy/parsing/latex/lark/grammar/latex.lark
@@ -96,7 +96,7 @@ UNDERSCORE: "_"
 
 // relational symbols
 EQUAL: "="
-NOT_EQUAL: "\\neq" | "\\ne"
+NOT_EQUAL: /\\neq(?![a-zA-Z])/ | /\\ne(?![a-zA-Z])/
 LT: "<"
 LTE: "\\leq" | "\\le" | "\\leqslant"
 GT: ">"
