diff --git a/doc/whatsnew/fragments/10960.false_positive b/doc/whatsnew/fragments/10960.false_positive
new file mode 100644
index 00000000000..3c0874c5c14
--- /dev/null
+++ b/doc/whatsnew/fragments/10960.false_positive
@@ -0,0 +1,3 @@
+Fix a false positive for ``bad-dunder-name`` when there is a user-defined ``__suppress_context__`` attribute on exception subclasses.
+
+Closes #10960
diff --git a/pylint/constants.py b/pylint/constants.py
index 6a2481913b8..b31099baf76 100644
--- a/pylint/constants.py
+++ b/pylint/constants.py
@@ -253,6 +253,7 @@ def _get_pylint_home() -> str:
     "__module__",
     "__sizeof__",
     "__subclasshook__",
+    "__suppress_context__",
     "__weakref__",
 ]
 
