diff --git a/tests/functional/ext/bad_dunder/bad_dunder_name.py b/tests/functional/ext/bad_dunder/bad_dunder_name.py
index 6008866b1d7..b576e8cbb97 100644
--- a/tests/functional/ext/bad_dunder/bad_dunder_name.py
+++ b/tests/functional/ext/bad_dunder/bad_dunder_name.py
@@ -53,5 +53,11 @@ def __index__(self):
         return 1
 
 
+class MyError(Exception):
+    @property
+    def __suppress_context__(self):
+        return True
+
+
 def __increase_me__(val):
     return val + 1
