diff --git a/xarray/tests/test_accessor_dt.py b/xarray/tests/test_accessor_dt.py
index de8181f6a28..7aff1c11250 100644
--- a/xarray/tests/test_accessor_dt.py
+++ b/xarray/tests/test_accessor_dt.py
@@ -58,7 +58,9 @@ def setup(self):
             "weekofyear",
             "dayofweek",
             "weekday",
+            "day_of_week",
             "dayofyear",
+            "day_of_year",
             "quarter",
             "date",
             "time",
@@ -77,20 +79,20 @@ def test_field_access(self, field) -> None:
         if field in ["week", "weekofyear"]:
             data = self.times.isocalendar()["week"]
         else:
-            data = getattr(self.times, field)
+            pandas_translations = {
+                "weekday": "day_of_week",
+                "dayofweek": "day_of_week",
+                "daysinmonth": "days_in_month",
+                "dayofyear": "day_of_year",
+            }
+            pandas_name = pandas_translations.get(field, field)
+            data = getattr(self.times, pandas_name)
 
         if data.dtype.kind != "b" and field not in ("date", "time"):
             # pandas 2.0 returns int32 for integer fields now
             data = data.astype("int64")
 
-        translations = {
-            "weekday": "dayofweek",
-            "daysinmonth": "days_in_month",
-            "weekofyear": "week",
-        }
-        name = translations.get(field, field)
-
-        expected = xr.DataArray(data, name=name, coords=[self.times], dims=["time"])
+        expected = xr.DataArray(data, name=field, coords=[self.times], dims=["time"])
 
         if field in ["week", "weekofyear"]:
             with pytest.warns(
@@ -179,7 +181,9 @@ def test_not_datetime_type(self) -> None:
             "weekofyear",
             "dayofweek",
             "weekday",
+            "day_of_week",
             "dayofyear",
+            "day_of_year",
             "quarter",
             "date",
             "time",
@@ -441,7 +445,7 @@ def times_3d(times):
 
 @requires_cftime
 @pytest.mark.parametrize(
-    "field", ["year", "month", "day", "hour", "dayofyear", "dayofweek"]
+    "field", ["year", "month", "day", "hour", "day_of_year", "day_of_week"]
 )
 def test_field_access(data, field) -> None:
     result = getattr(data.time.dt, field)
@@ -533,7 +537,7 @@ def test_cftime_strftime_access(data) -> None:
 @requires_cftime
 @requires_dask
 @pytest.mark.parametrize(
-    "field", ["year", "month", "day", "hour", "dayofyear", "dayofweek"]
+    "field", ["year", "month", "day", "hour", "day_of_year", "day_of_week"]
 )
 def test_dask_field_access_1d(data, field) -> None:
     import dask.array as da
@@ -553,7 +557,7 @@ def test_dask_field_access_1d(data, field) -> None:
 @requires_cftime
 @requires_dask
 @pytest.mark.parametrize(
-    "field", ["year", "month", "day", "hour", "dayofyear", "dayofweek"]
+    "field", ["year", "month", "day", "hour", "day_of_year", "day_of_week"]
 )
 def test_dask_field_access(times_3d, data, field) -> None:
     import dask.array as da
diff --git a/xarray/tests/test_calendar_ops.py b/xarray/tests/test_calendar_ops.py
index 927bebd3f5a..87093bf0786 100644
--- a/xarray/tests/test_calendar_ops.py
+++ b/xarray/tests/test_calendar_ops.py
@@ -159,7 +159,7 @@ def test_convert_calendar_360_days_random():
     # Ensure that added days are evenly distributed in the 5 fifths of each year
     conv = convert_calendar(da_360, "noleap", align_on="random", missing=np.nan)
     conv = conv.where(conv.isnull(), drop=True)
-    nandoys = conv.time.dt.dayofyear[:366]
+    nandoys = conv.time.dt.day_of_year[:366]
     assert all(nandoys < np.array([74, 147, 220, 293, 366]))
     assert all(nandoys > np.array([0, 73, 146, 219, 292]))
 
diff --git a/xarray/tests/test_cftime_offsets.py b/xarray/tests/test_cftime_offsets.py
index 8dfb357ebca..5011f1e788b 100644
--- a/xarray/tests/test_cftime_offsets.py
+++ b/xarray/tests/test_cftime_offsets.py
@@ -1380,19 +1380,19 @@ def test_calendar_year_length(
 
 @pytest.mark.parametrize("freq", ["YE", "ME", "D"])
 def test_dayofweek_after_cftime(freq: str) -> None:
-    result = date_range("2000-02-01", periods=3, freq=freq, use_cftime=True).dayofweek
+    result = date_range("2000-02-01", periods=3, freq=freq, use_cftime=True).day_of_week
     # TODO: remove once requiring pandas 2.2+
     freq = _new_to_legacy_freq(freq)
-    expected = pd.date_range("2000-02-01", periods=3, freq=freq).dayofweek
+    expected = pd.date_range("2000-02-01", periods=3, freq=freq).day_of_week
     np.testing.assert_array_equal(result, expected)
 
 
 @pytest.mark.parametrize("freq", ["YE", "ME", "D"])
 def test_dayofyear_after_cftime(freq: str) -> None:
-    result = date_range("2000-02-01", periods=3, freq=freq, use_cftime=True).dayofyear
+    result = date_range("2000-02-01", periods=3, freq=freq, use_cftime=True).day_of_year
     # TODO: remove once requiring pandas 2.2+
     freq = _new_to_legacy_freq(freq)
-    expected = pd.date_range("2000-02-01", periods=3, freq=freq).dayofyear
+    expected = pd.date_range("2000-02-01", periods=3, freq=freq).day_of_year
     np.testing.assert_array_equal(result, expected)
 
 
diff --git a/xarray/tests/test_cftimeindex.py b/xarray/tests/test_cftimeindex.py
index 8aa79be458e..7f6f391fa6b 100644
--- a/xarray/tests/test_cftimeindex.py
+++ b/xarray/tests/test_cftimeindex.py
@@ -315,8 +315,8 @@ def test_cftimeindex_field_accessors(index, field, expected):
         "minute",
         "second",
         "microsecond",
-        "dayofyear",
-        "dayofweek",
+        "day_of_year",
+        "day_of_week",
         "days_in_month",
     ],
 )
@@ -329,16 +329,18 @@ def test_empty_cftimeindex_field_accessors(field):
 
 
 @requires_cftime
-def test_cftimeindex_dayofyear_accessor(index):
-    result = index.dayofyear
+@pytest.mark.parametrize("field", ["day_of_year", "dayofyear"])
+def test_cftimeindex_dayofyear_accessor(index, field):
+    result = getattr(index, field)
     expected = np.array([date.dayofyr for date in index], dtype=np.int64)
     assert_array_equal(result, expected)
     assert result.dtype == expected.dtype
 
 
 @requires_cftime
-def test_cftimeindex_dayofweek_accessor(index):
-    result = index.dayofweek
+@pytest.mark.parametrize("field", ["day_of_week", "dayofweek"])
+def test_cftimeindex_dayofweek_accessor(index, field):
+    result = getattr(index, field)
     expected = np.array([date.dayofwk for date in index], dtype=np.int64)
     assert_array_equal(result, expected)
     assert result.dtype == expected.dtype
diff --git a/xarray/tests/test_cupy.py b/xarray/tests/test_cupy.py
index 94776902c11..29b7055ca30 100644
--- a/xarray/tests/test_cupy.py
+++ b/xarray/tests/test_cupy.py
@@ -21,7 +21,7 @@ def toy_weather_data():
     """
     np.random.seed(123)
     times = pd.date_range("2000-01-01", "2001-12-31", name="time")
-    annual_cycle = np.sin(2 * np.pi * (times.dayofyear.values / 365.25 - 0.28))
+    annual_cycle = np.sin(2 * np.pi * (times.day_of_year.values / 365.25 - 0.28))
 
     base = 10 + 15 * annual_cycle.reshape(-1, 1)
     tmin_values = base + 3 * np.random.randn(annual_cycle.size, 3)
diff --git a/xarray/tests/test_dataarray.py b/xarray/tests/test_dataarray.py
index 8eb52046a31..377fd2f8a8b 100644
--- a/xarray/tests/test_dataarray.py
+++ b/xarray/tests/test_dataarray.py
@@ -1514,8 +1514,8 @@ def test_virtual_time_components(self) -> None:
         dates = pd.date_range("2000-01-01", periods=10)
         da = DataArray(np.arange(1, 11), [("time", dates)])
 
-        assert_array_equal(da["time.dayofyear"], da.values)
-        assert_array_equal(da.coords["time.dayofyear"], da.values)
+        assert_array_equal(da["time.day_of_year"], da.values)
+        assert_array_equal(da.coords["time.day_of_year"], da.values)
 
     def test_coords(self) -> None:
         # use int64 to ensure repr() consistency on windows
diff --git a/xarray/tests/test_dataset.py b/xarray/tests/test_dataset.py
index cd4f2c10eb2..762c647f15c 100644
--- a/xarray/tests/test_dataset.py
+++ b/xarray/tests/test_dataset.py
@@ -1737,7 +1737,7 @@ def test_sel(self) -> None:
         times = pd.date_range("2000-01-01", periods=3)
         assert_equal(data.isel(time=slice(3)), data.sel(time=times))
         assert_equal(
-            data.isel(time=slice(3)), data.sel(time=(data["time.dayofyear"] <= 3))
+            data.isel(time=slice(3)), data.sel(time=(data["time.day_of_year"] <= 3))
         )
 
         td = pd.to_timedelta(np.arange(3), unit="days")
@@ -4645,11 +4645,11 @@ def test_virtual_variables_time(self) -> None:
         assert_array_equal(data["time.month"].values, index.month)
         assert_array_equal(data["time.season"].values, "DJF")
         # test virtual variable math
-        assert_array_equal(data["time.dayofyear"] + 1, 2 + np.arange(20))
-        assert_array_equal(np.sin(data["time.dayofyear"]), np.sin(1 + np.arange(20)))
+        assert_array_equal(data["time.day_of_year"] + 1, 2 + np.arange(20))
+        assert_array_equal(np.sin(data["time.day_of_year"]), np.sin(1 + np.arange(20)))
         # ensure they become coordinates
-        expected = Dataset({}, {"dayofyear": data["time.dayofyear"]})
-        actual = data[["time.dayofyear"]]
+        expected = Dataset({}, {"day_of_year": data["time.day_of_year"]})
+        actual = data[["time.day_of_year"]]
         assert_equal(expected, actual)
         # non-coordinate variables
         ds = Dataset({"t": ("x", pd.date_range("2000-01-01", periods=3))})
@@ -4672,9 +4672,10 @@ def test_time_season(self) -> None:
     def test_slice_virtual_variable(self) -> None:
         data = create_test_data()
         assert_equal(
-            data["time.dayofyear"][:10].variable, Variable(["time"], 1 + np.arange(10))
+            data["time.day_of_year"][:10].variable,
+            Variable(["time"], 1 + np.arange(10)),
         )
-        assert_equal(data["time.dayofyear"][0].variable, Variable([], 1))
+        assert_equal(data["time.day_of_year"][0].variable, Variable([], 1))
 
     def test_setitem(self) -> None:
         # assign a variable
diff --git a/xarray/tests/test_formatting_html.py b/xarray/tests/test_formatting_html.py
index 8f1358b755f..5160cc29066 100644
--- a/xarray/tests/test_formatting_html.py
+++ b/xarray/tests/test_formatting_html.py
@@ -95,7 +95,7 @@ def multiindex() -> xr.Dataset:
 @pytest.fixture
 def dataset() -> xr.Dataset:
     times = pd.date_range("2000-01-01", "2001-12-31", name="time")
-    annual_cycle = np.sin(2 * np.pi * (times.dayofyear.values / 365.25 - 0.28))
+    annual_cycle = np.sin(2 * np.pi * (times.day_of_year.values / 365.25 - 0.28))
 
     base = 10 + 15 * annual_cycle.reshape(-1, 1)
     tmin_values = base + 3 * np.random.randn(annual_cycle.size, 3)
diff --git a/xarray/tests/test_groupby.py b/xarray/tests/test_groupby.py
index 7c06f7c1525..0a5900f4f40 100644
--- a/xarray/tests/test_groupby.py
+++ b/xarray/tests/test_groupby.py
@@ -2921,7 +2921,7 @@ def reset(self):
 def test_weather_data_resample(use_flox):
     # from the docs
     times = pd.date_range("2000-01-01", "2001-12-31", name="time")
-    annual_cycle = np.sin(2 * np.pi * (times.dayofyear.values / 365.25 - 0.28))
+    annual_cycle = np.sin(2 * np.pi * (times.day_of_year.values / 365.25 - 0.28))
 
     base = 10 + 15 * annual_cycle.reshape(-1, 1)
     tmin_values = base + 3 * np.random.randn(annual_cycle.size, 3)
