diff --git a/django/contrib/postgres/constraints.py b/django/contrib/postgres/constraints.py
index ad3a5f61f5..75e3b3378f 100644
--- a/django/contrib/postgres/constraints.py
+++ b/django/contrib/postgres/constraints.py
@@ -33,6 +33,7 @@ class ExclusionConstraint(BaseConstraint):
         deferrable=None,
         include=None,
         violation_error_message=None,
+        violation_error_code=None,
     ):
         if index_type and index_type.lower() not in {"gist", "spgist"}:
             raise ValueError(
@@ -60,7 +61,11 @@ class ExclusionConstraint(BaseConstraint):
         self.condition = condition
         self.deferrable = deferrable
         self.include = tuple(include) if include else ()
-        super().__init__(name=name, violation_error_message=violation_error_message)
+        super().__init__(
+            name=name,
+            violation_error_message=violation_error_message,
+            violation_error_code=violation_error_code,
+        )
 
     def _get_expressions(self, schema_editor, query):
         expressions = []
@@ -150,6 +155,7 @@ class ExclusionConstraint(BaseConstraint):
                 and self.deferrable == other.deferrable
                 and self.include == other.include
                 and self.violation_error_message == other.violation_error_message
+                and self.violation_error_code == other.violation_error_code
             )
         return super().__eq__(other)
 
@@ -167,6 +173,12 @@ class ExclusionConstraint(BaseConstraint):
                 if self.violation_error_message is None
                 or self.violation_error_message == self.default_violation_error_message
                 else " violation_error_message=%r" % self.violation_error_message
+            )
+            + (
+                ""
+                if self.violation_error_code is None
+                or self.violation_error_code == self.default_violation_error_code
+                else " violation_error_code=%r" % self.violation_error_code
             ),
         )
 
@@ -204,9 +216,15 @@ class ExclusionConstraint(BaseConstraint):
             queryset = queryset.exclude(pk=model_class_pk)
         if not self.condition:
             if queryset.exists():
-                raise ValidationError(self.get_violation_error_message())
+                raise ValidationError(
+                    self.get_violation_error_message(),
+                    code=self.get_violation_error_code(),
+                )
         else:
             if (self.condition & Exists(queryset.filter(self.condition))).check(
                 replacement_map, using=using
             ):
-                raise ValidationError(self.get_violation_error_message())
+                raise ValidationError(
+                    self.get_violation_error_message(),
+                    code=self.get_violation_error_code(),
+                )
diff --git a/django/db/models/constraints.py b/django/db/models/constraints.py
index 075ecee1be..8194bd0dda 100644
--- a/django/db/models/constraints.py
+++ b/django/db/models/constraints.py
@@ -19,10 +19,16 @@ __all__ = ["BaseConstraint", "CheckConstraint", "Deferrable", "UniqueConstraint"
 class BaseConstraint:
     default_violation_error_message = _("Constraint “%(name)s” is violated.")
     violation_error_message = None
+    default_violation_error_code = None
+    violation_error_code = None
 
     # RemovedInDjango60Warning: When the deprecation ends, replace with:
-    # def __init__(self, *, name, violation_error_message=None):
-    def __init__(self, *args, name=None, violation_error_message=None):
+    # def __init__(
+    #     self, *, name, violation_error_message=None, violation_error_code=None
+    # ):
+    def __init__(
+        self, *args, name=None, violation_error_message=None, violation_error_code=None
+    ):
         # RemovedInDjango60Warning.
         if name is None and not args:
             raise TypeError(
@@ -34,6 +40,10 @@ class BaseConstraint:
             self.violation_error_message = violation_error_message
         else:
             self.violation_error_message = self.default_violation_error_message
+        if violation_error_code is not None:
+            self.violation_error_code = violation_error_code
+        else:
+            self.violation_error_code = self.default_violation_error_code
         # RemovedInDjango60Warning.
         if args:
             warnings.warn(
@@ -65,6 +75,9 @@ class BaseConstraint:
     def get_violation_error_message(self):
         return self.violation_error_message % {"name": self.name}
 
+    def get_violation_error_code(self):
+        return self.violation_error_code
+
     def deconstruct(self):
         path = "%s.%s" % (self.__class__.__module__, self.__class__.__name__)
         path = path.replace("django.db.models.constraints", "django.db.models")
@@ -74,6 +87,11 @@ class BaseConstraint:
             and self.violation_error_message != self.default_violation_error_message
         ):
             kwargs["violation_error_message"] = self.violation_error_message
+        if (
+            self.violation_error_code is not None
+            and self.violation_error_code != self.default_violation_error_code
+        ):
+            kwargs["violation_error_code"] = self.violation_error_code
         return (path, (), kwargs)
 
     def clone(self):
@@ -82,13 +100,19 @@ class BaseConstraint:
 
 
 class CheckConstraint(BaseConstraint):
-    def __init__(self, *, check, name, violation_error_message=None):
+    def __init__(
+        self, *, check, name, violation_error_message=None, violation_error_code=None
+    ):
         self.check = check
         if not getattr(check, "conditional", False):
             raise TypeError(
                 "CheckConstraint.check must be a Q instance or boolean expression."
             )
-        super().__init__(name=name, violation_error_message=violation_error_message)
+        super().__init__(
+            name=name,
+            violation_error_message=violation_error_message,
+            violation_error_code=violation_error_code,
+        )
 
     def _get_check_sql(self, model, schema_editor):
         query = Query(model=model, alias_cols=False)
@@ -112,7 +136,10 @@ class CheckConstraint(BaseConstraint):
         against = instance._get_field_value_map(meta=model._meta, exclude=exclude)
         try:
             if not Q(self.check).check(against, using=using):
-                raise ValidationError(self.get_violation_error_message())
+                raise ValidationError(
+                    self.get_violation_error_message(),
+                    code=self.get_violation_error_code(),
+                )
         except FieldError:
             pass
 
@@ -126,6 +153,12 @@ class CheckConstraint(BaseConstraint):
                 if self.violation_error_message is None
                 or self.violation_error_message == self.default_violation_error_message
                 else " violation_error_message=%r" % self.violation_error_message
+            )
+            + (
+                ""
+                if self.violation_error_code is None
+                or self.violation_error_code == self.default_violation_error_code
+                else " violation_error_code=%r" % self.violation_error_code
             ),
         )
 
@@ -135,6 +168,7 @@ class CheckConstraint(BaseConstraint):
                 self.name == other.name
                 and self.check == other.check
                 and self.violation_error_message == other.violation_error_message
+                and self.violation_error_code == other.violation_error_code
             )
         return super().__eq__(other)
 
@@ -164,6 +198,7 @@ class UniqueConstraint(BaseConstraint):
         include=None,
         opclasses=(),
         violation_error_message=None,
+        violation_error_code=None,
     ):
         if not name:
             raise ValueError("A unique constraint must be named.")
@@ -213,7 +248,11 @@ class UniqueConstraint(BaseConstraint):
             F(expression) if isinstance(expression, str) else expression
             for expression in expressions
         )
-        super().__init__(name=name, violation_error_message=violation_error_message)
+        super().__init__(
+            name=name,
+            violation_error_message=violation_error_message,
+            violation_error_code=violation_error_code,
+        )
 
     @property
     def contains_expressions(self):
@@ -307,6 +346,12 @@ class UniqueConstraint(BaseConstraint):
                 if self.violation_error_message is None
                 or self.violation_error_message == self.default_violation_error_message
                 else " violation_error_message=%r" % self.violation_error_message
+            )
+            + (
+                ""
+                if self.violation_error_code is None
+                or self.violation_error_code == self.default_violation_error_code
+                else " violation_error_code=%r" % self.violation_error_code
             ),
         )
 
@@ -321,6 +366,7 @@ class UniqueConstraint(BaseConstraint):
                 and self.opclasses == other.opclasses
                 and self.expressions == other.expressions
                 and self.violation_error_message == other.violation_error_message
+                and self.violation_error_code == other.violation_error_code
             )
         return super().__eq__(other)
 
@@ -385,7 +431,10 @@ class UniqueConstraint(BaseConstraint):
         if not self.condition:
             if queryset.exists():
                 if self.expressions:
-                    raise ValidationError(self.get_violation_error_message())
+                    raise ValidationError(
+                        self.get_violation_error_message(),
+                        code=self.get_violation_error_code(),
+                    )
                 # When fields are defined, use the unique_error_message() for
                 # backward compatibility.
                 for model, constraints in instance.get_constraints():
@@ -400,6 +449,9 @@ class UniqueConstraint(BaseConstraint):
                 if (self.condition & Exists(queryset.filter(self.condition))).check(
                     against, using=using
                 ):
-                    raise ValidationError(self.get_violation_error_message())
+                    raise ValidationError(
+                        self.get_violation_error_message(),
+                        code=self.get_violation_error_code(),
+                    )
             except FieldError:
                 pass
diff --git a/docs/ref/contrib/postgres/constraints.txt b/docs/ref/contrib/postgres/constraints.txt
index fcf50b8b5f..4301085c3a 100644
--- a/docs/ref/contrib/postgres/constraints.txt
+++ b/docs/ref/contrib/postgres/constraints.txt
@@ -12,7 +12,7 @@ PostgreSQL supports additional data integrity constraints available from the
 ``ExclusionConstraint``
 =======================
 
-.. class:: ExclusionConstraint(*, name, expressions, index_type=None, condition=None, deferrable=None, include=None, violation_error_message=None)
+.. class:: ExclusionConstraint(*, name, expressions, index_type=None, condition=None, deferrable=None, include=None, violation_error_message=None, violation_error_code=None)
 
     Creates an exclusion constraint in the database. Internally, PostgreSQL
     implements exclusion constraints using indexes. The default index type is
@@ -140,6 +140,13 @@ The error message used when ``ValidationError`` is raised during
 :ref:`model validation <validating-objects>`. Defaults to
 :attr:`.BaseConstraint.violation_error_message`.
 
+``violation_error_code``
+------------------------
+
+The error code used when ``ValidationError`` is raised during
+:ref:`model validation <validating-objects>`. Defaults to
+:attr:`.BaseConstraint.violation_error_code`.
+
 Examples
 --------
 
diff --git a/docs/ref/models/constraints.txt b/docs/ref/models/constraints.txt
index baaacd8754..bde1a1bfc6 100644
--- a/docs/ref/models/constraints.txt
+++ b/docs/ref/models/constraints.txt
@@ -48,7 +48,7 @@ option.
 ``BaseConstraint``
 ==================
 
-.. class:: BaseConstraint(*, name, violation_error_message=None)
+.. class:: BaseConstraint(*, name, violation_error_message=None, violation_error_code=None)
 
     Base class for all constraints. Subclasses must implement
     ``constraint_sql()``, ``create_sql()``, ``remove_sql()`` and
@@ -77,6 +77,14 @@ The error message used when ``ValidationError`` is raised during
 :ref:`model validation <validating-objects>`. Defaults to
 ``"Constraint “%(name)s” is violated."``.
 
+``violation_error_code``
+------------------------
+
+.. attribute:: BaseConstraint.violation_error_code
+
+The error code used when ``ValidationError`` is raised during
+:ref:`model validation <validating-objects>`. Defaults to ``None``.
+
 ``validate()``
 --------------
 
@@ -94,7 +102,7 @@ This method must be implemented by a subclass.
 ``CheckConstraint``
 ===================
 
-.. class:: CheckConstraint(*, check, name, violation_error_message=None)
+.. class:: CheckConstraint(*, check, name, violation_error_message=None, violation_error_code=None)
 
     Creates a check constraint in the database.
 
@@ -121,7 +129,7 @@ ensures the age field is never less than 18.
 ``UniqueConstraint``
 ====================
 
-.. class:: UniqueConstraint(*expressions, fields=(), name=None, condition=None, deferrable=None, include=None, opclasses=(), violation_error_message=None)
+.. class:: UniqueConstraint(*expressions, fields=(), name=None, condition=None, deferrable=None, include=None, opclasses=(), violation_error_message=None, violation_error_code=None)
 
     Creates a unique constraint in the database.
 
@@ -257,3 +265,19 @@ This message is *not used* for :class:`UniqueConstraint`\s with
 same message as constraints defined with
 :attr:`.Field.unique` or in
 :attr:`Meta.unique_together <django.db.models.Options.constraints>`.
+
+``violation_error_code``
+------------------------
+
+.. attribute:: UniqueConstraint.violation_error_code
+
+The error code used when ``ValidationError`` is raised during
+:ref:`model validation <validating-objects>`. Defaults to
+:attr:`.BaseConstraint.violation_error_code`.
+
+This code is *not used* for :class:`UniqueConstraint`\s with
+:attr:`~UniqueConstraint.fields` and without a
+:attr:`~UniqueConstraint.condition`. Such :class:`~UniqueConstraint`\s show the
+same message as constraints defined with
+:attr:`.Field.unique` or in
+:attr:`Meta.unique_together <django.db.models.Options.constraints>`.
diff --git a/docs/releases/5.0.txt b/docs/releases/5.0.txt
index 14e767cfd8..c6799c68f5 100644
--- a/docs/releases/5.0.txt
+++ b/docs/releases/5.0.txt
@@ -182,6 +182,10 @@ Models
   and :meth:`.QuerySet.aupdate_or_create` methods allows specifying a different
   field values for the create operation.
 
+* The new ``violation_error_code`` argument of
+  :class:`~django.db.models.BaseConstraint` allows specifying the error code
+  used when a ``ValidationError`` is raised during model validation.
+
 Requests and Responses
 ~~~~~~~~~~~~~~~~~~~~~~
 
