diff --git a/sympy/polys/tests/test_densearith.py b/sympy/polys/tests/test_densearith.py
index c52c1009987d..c8fae3cc1464 100644
--- a/sympy/polys/tests/test_densearith.py
+++ b/sympy/polys/tests/test_densearith.py
@@ -190,6 +190,8 @@ def test_dup_mul_ground():
 
 
 def test_dmp_mul_ground():
+    assert dmp_mul_ground(f_0, ZZ(0), 2, ZZ) == [[[]]]
+
     assert dmp_mul_ground(f_0, ZZ(2), 2, ZZ) == [
         [[ZZ(2), ZZ(4), ZZ(6)], [ZZ(4)]],
         [[ZZ(6)]],
diff --git a/sympy/polys/tests/test_polytools.py b/sympy/polys/tests/test_polytools.py
index 2f9436efb0b7..5847f50753d2 100644
--- a/sympy/polys/tests/test_polytools.py
+++ b/sympy/polys/tests/test_polytools.py
@@ -740,6 +740,12 @@ def test_Poly_mul_ground():
     assert Poly(x + 1).mul_ground(2) == Poly(2*x + 2)
 
 
+def test_Poly_rep_mul_zero_is_zero():
+    # issue 29715
+    p = Poly(x**4 + y, x, y)
+    assert (p.rep * 0).is_zero is True
+
+
 def test_Poly_quo_ground():
     assert Poly(2*x + 4).quo_ground(2) == Poly(x + 2)
     assert Poly(2*x + 3).quo_ground(2) == Poly(x + 1)
@@ -1663,6 +1669,10 @@ def test_Poly_diff():
     assert Poly(x**2*y**2 + x*y).diff(x, y) == Poly(4*x*y + 1)
     assert Poly(x**2*y**2 + x*y).diff(y, x) == Poly(4*x*y + 1)
 
+    p = Poly(x**7*y**5 + 2*x**7*y**4 + x**2, x, y, z, modulus=7)
+    assert p.diff() == Poly(2*x, x, y, z, modulus=7)
+    assert p.diff().gcd(p) == Poly(x, x, y, z, modulus=7)
+
 
 def test_issue_9585():
     assert diff(Poly(x**2 + x)) == Poly(2*x + 1)
