diff --git a/.mailmap b/.mailmap
index 206d8610ca25..2d5a6b65c480 100644
--- a/.mailmap
+++ b/.mailmap
@@ -1861,6 +1861,7 @@ mao8 <thisisma08@gmail.com>
 mohajain <mohajain99@gmail.com> mohajain <45903778+mohajain@users.noreply.github.com>
 mohammedouahman <simofun85@gmail.com>
 mohit <39158356+mohitacecode@users.noreply.github.com> mohit <42018918+mohitshah3111999@users.noreply.github.com>
+morluto <76467478+morluto@users.noreply.github.com>
 naelsondouglas <naelson17@gmail.com>
 noam simcha finkelstein <noam.finkelstein@protonmail.com>
 numbermaniac <5206120+numbermaniac@users.noreply.github.com>
diff --git a/sympy/polys/subresultants_qq_zz.py b/sympy/polys/subresultants_qq_zz.py
index 3f1782d3c45c..e4c2b9089745 100644
--- a/sympy/polys/subresultants_qq_zz.py
+++ b/sympy/polys/subresultants_qq_zz.py
@@ -1431,6 +1431,10 @@ def modified_subresultants_pg(p, q, x):
     # first remainder
     exp_deg = d1 - 1                          # expected degree of a2
     a2 = - rem(a0, a1, domain=QQ)             # first remainder
+
+    if a2 == 0:                               # q | p; sequence terminates
+        return subres_l                       # with the pair (p, q)
+
     rho2 =  LC(a2, x)                         # leading coeff of a2
     d2 =  degree(a2, x)                       # actual degree of a2
     deg_diff_new = exp_deg - d2               # expected - actual degree
@@ -1483,6 +1487,10 @@ def modified_subresultants_pg(p, q, x):
         del0 = del1                          # update degree difference
         exp_deg = d1 - 1                     # new expected degree
         a2 = - rem(a0, a1, domain=QQ)        # new remainder
+
+        if a2 == 0:                          # exact division; sequence complete
+            break
+
         rho3 =  LC(a2, x)                    # leading coeff of a2
         d2 =  degree(a2, x)                  # actual degree of a2
         deg_diff_new = exp_deg - d2          # expected - actual degree
